Mathlib/Data/Nat/WithBot.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+
+! This file was ported from Lean 3 source module data.nat.with_bot
+! leanprover-community/mathlib commit 46a64b5b4268c594af770c44d9e502afc6a515cb
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+
+import Mathlib.Data.Nat.Order.Basic
+import Mathlib.Algebra.Order.Monoid.WithTop
+import Aesop
+
+/-!
+# `with_bot ℕ`
+
+Lemmas about the type of natural numbers with a bottom element adjoined.
+-/
+
+
+namespace Nat
+
+namespace WithBot
+
+theorem add_eq_zero_iff {n m : WithBot ℕ} : n + m = 0 ↔ n = 0 ∧ m = 0 :=
+  by
```

Comment:
```suggestion
theorem add_eq_zero_iff {n m : WithBot ℕ} : n + m = 0 ↔ n = 0 ∧ m = 0 := by
```
This is the style we're going for. (And the same for the rest of the file)

Reply:
done